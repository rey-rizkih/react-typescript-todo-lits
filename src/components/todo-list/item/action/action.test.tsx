import { shallow } from "enzyme";
import TodoAction, { TodoActionProps } from ".";

let wrapper;

const mockActionProps: TodoActionProps = {
  onDelete: jest.fn(),
  onEdit: jest.fn(),
  onDone: jest.fn(),
};

const testId = {
  actionContainer: '[data-testid="action-container"]',
  editButton: '[data-testid="edit-button"]',
  deleteButton: '[data-testid="delete-button"]',
  doneButton: '[data-testid="check-button"]',
};

describe("<TodoAction/> rendering", () => {
  beforeEach(() => {
    wrapper = shallow(<TodoAction {...mockActionProps} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render one", () => {
    expect(wrapper.find(testId.actionContainer)).toHaveLength(1);
  });

  it("should render <IconButton/> for each todo", () => {
    expect(wrapper.find(testId.editButton)).toHaveLength(1);
    expect(wrapper.find(testId.deleteButton)).toHaveLength(1);
    expect(wrapper.find(testId.doneButton)).toHaveLength(1);
  });

  describe("<TodoAction/> interactions", () => {
    it("should call onDelete when delete button clicked", () => {
      wrapper.find(testId.deleteButton).simulate("click");

      expect(mockActionProps.onDelete).toHaveBeenCalled();
    });

    it("should call onEdit when edit button clicked", () => {
      wrapper.find(testId.editButton).simulate("click");

      expect(mockActionProps.onEdit).toHaveBeenCalled();
    });

    it("should call onDone when done button clicked", () => {
      wrapper.find(testId.doneButton).simulate("click");

      expect(mockActionProps.onDone).toHaveBeenCalled();
    });
  });
});
