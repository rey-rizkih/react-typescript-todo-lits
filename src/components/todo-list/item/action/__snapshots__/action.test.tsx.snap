// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoAction/> rendering renders correctly 1`] = `
<Styled(div)
  data-testid="action-container"
>
  <ForwardRef(IconButton)
    aria-label="edit todo"
    color="inherit"
    data-testid="edit-button"
    onClick={[MockFunction]}
  >
    <Memo(ForwardRef(EditIcon)) />
  </ForwardRef(IconButton)>
  <ForwardRef(IconButton)
    aria-label="delete todo"
    color="inherit"
    data-testid="delete-button"
    onClick={[MockFunction]}
  >
    <Memo(ForwardRef(DeleteIcon)) />
  </ForwardRef(IconButton)>
  <ForwardRef(IconButton)
    aria-label="check todo"
    color="inherit"
    data-testid="check-button"
    onClick={[MockFunction]}
  >
    <Memo(ForwardRef(CheckIcon)) />
  </ForwardRef(IconButton)>
</Styled(div)>
`;
